### ISSUE TITLE
Find duplicates also inside archives (zip, 7z, tar.gz, bz2, etc)

### ISSUE THREAD

**Issue by sergeyvolk** [qarmin/czkawka#616]
Thanks, this is a very useful tool!
I think it would be even more useful if it had an option to look for duplicate files also inside archives. In my personal archives I have a lot of files stored as uncompressed files (recent additions) and older files organized into many compressed .zip files. Currently if I want to find duplicates, I need to go unzip all those .zips, which is technically possible, but inconvenient.

**Comment by BloodBlight** [comment]
I would also love to see this!

**Comment by mpissarello** [comment]
+1 interested in this feature

**Comment by 0cwa** [comment]
I would also find this extremely helpful -- especially filesystem images ( .iso .img etc..) and the squashfs images from inside of .appimages.  Recently I rm -rf 'd a big chunk of my home folder, so I shut down and ran photorec (binary-carving) and xfs_undelete.  I had backups but I can't automatically remove all these extracted application files from the fs recovery, so now I'm left with 100s of gb of files, much of which have been taken out filesystem images or perhaps binary carved uncompressed archives.

I could not find a linux duplicate finder which implements this feature, so it would be extra useful to have a duplicate finder which implements this 🙏

**Comment by motylewski** [comment]
Yes, I vote for that feature. A lot of my duplicate files and directories are stored as compressed archives or filesystem images (sometimes single partitions, sometimes the whole disk image including partition table)

Of course I could decompress such archives or images using some script to a temporary location or even temporarily mount them and let czkawka scan them. For example in addition to archive.zip there would be a (virtual) directory archive.zip_contents including the files. This could be even recursive.

EDIT: it looks that mounting most archive formats is done there: https://github.com/mxmlnkn/ratarmount

**Comment by 0cwa** [comment]
I've been working on my own version of something like this here: https://github.com/0cwa/Dedupe-Archived-Files
